src/wizards/tokenSwapDealWizard/stages/termsStage/termsStage.ts

```diff
@@ -2,25 +2,31 @@ import { IBaseWizardStage, IStageMeta } from "../../dealWizardTypes";
 import { autoinject } from "aurelia-framework";
 import { IWizardState, WizardService } from "../../../services/WizardService";
 import { IClause, IDealRegistrationTokenSwap } from "entities/DealRegistrationTokenSwap";
-import { ValidationController } from "aurelia-validation";
 import "./termsStage.scss";
 import { areFormsValid } from "../../../../services/ValidationService";
+import { TermClause } from "./termClause/termClause";
 
 @autoinject
 export class TermsStage implements IBaseWizardStage {
   public wizardManager: any;
   public wizardState: IWizardState<IDealRegistrationTokenSwap>;
 
-  private termsForms: ValidationController[] = [];
+  termClauses: TermClause[] = [];
+  hasUnsavedChanges = false;
 
-  constructor(public wizardService: WizardService) {
+  constructor(
+    public wizardService: WizardService,
+  ) {
   }
 
   activate(stageMeta: IStageMeta): void {
     this.wizardManager = stageMeta.wizardManager;
     this.wizardState = this.wizardService.getWizardState(this.wizardManager);
 
-    this.wizardService.registerStageValidateFunction(this.wizardManager, () => areFormsValid(this.termsForms));
+    this.wizardService.registerStageValidateFunction(this.wizardManager, async () => {
+      this.checkedForUnsavedChanges();
+      return await areFormsValid(this.termClauses.map(viewModel => viewModel.form)) && !this.hasUnsavedChanges;
```

Comment:
remove `await` and `async`

Reply:
it is needed in order to make the expression return a boolean with the expected values

Reply:
Sorry you are right, I wasn't accounting for the rest of the expression.